spec/client/ics-007-tendermint-client/README.md

```diff
@@ -278,22 +281,28 @@ function checkForMisbehaviour(
           return true
         }
       case Misbehaviour:
-        // assert that the heights are the same
-        assert(misbehaviour.h1.height === misbehaviour.h2.height)
-        // assert that the commitments are different
-        assert(misbehaviour.h1.commitmentRoot !== misbehaviour.h2.commitmentRoot)
+        // if heights are equal check that this is valid misbehaviour of a fork
+        if (misbehaviour.h1.height === misbehaviour.h2.height && misbehaviour.h1.commitmentRoot !== misbehaviour.h2.commitmentRoot) {
+          return true
+        }
+        // otherwise if heights are unequal check that this is valid misbehavior of BFT time violation
+        if (misbehaviour.h1.timestamp <= misbehaviour.h2.timestamp) {
```

Comment:
This check assumes that `h1.height > h2.height`, but I am not sure if we should also check the other way around (i.e. `h2.height > h1.height && h2.timestamp <= misbehaviour.h1.timestamp`). [In ibc-go we also seem to assume that `h1.height > h2.height`...](https://github.com/cosmos/ibc-go/blob/7d60efa07b7d564e5d4f531a0398d3f670fc973f/modules/light-clients/07-tendermint/misbehaviour_handle.go#L89).

Reply:
See [comment](https://github.com/cosmos/ibc/pull/907/files#r1089430849).